app/src/main/java/com/example/next_show/data/RecommendationClient.java

```diff
@@ -25,13 +26,37 @@
     public static final int FORBIDDEN_REQUEST = 403;
     public static final int LIMIT = 5;
 
+    // instance vars
+    private Shows showsObj;
+    private FeedFragment.GenreMatchedCallback genreMatchedCallback;
+    private FeedFragment.ShowCallback showCallback;
+
     // empty constructor
     public RecommendationClient() { }
 
-    public void fetchRelatedShows(Shows traktShows, List<String> savedShows, ResponseCallback callback) {
+    public RecommendationClient(Shows showsObj) {
+        this.showsObj = showsObj;
+    }
+
+    public void fetchRecommendedShows(List<String> savedShows, FeedFragment.ShowCallback showCallback, FeedFragment.GenreMatchedCallback gmCallback) {
+        this.genreMatchedCallback = gmCallback;
+        this.showCallback = showCallback;
+
+        if (savedShows.isEmpty()){
+            // if no shows to compare, call user genre matched ones
+            fetchGenreMatchedShows(showsObj, genreMatchedCallback);
+            return;
+        }
+
+        // get related shows based on saved LIKED shows -> updates adapter within callback
+        fetchRelatedShows(showsObj, savedShows, showCallback);
```

Comment:
You should be able to separate this out into 2 separate method calls and then decide in the feed fragment which to call.  That would be a cleaner API